src/Utilities/Compiler/DoNotCatchGeneralUnlessRethrown.cs

```diff
@@ -78,6 +84,18 @@ public override void Initialize(AnalysisContext analysisContext)
                         IsConfiguredDisallowedExceptionType(type, method, compilationStartAnalysisContext.Compilation,
                             compilationStartAnalysisContext.Options, compilationStartAnalysisContext.CancellationToken);
 
+                    bool IsExcludedSymbol(SyntaxTree tree)
+                    {
+                        if (!_allowExcludedSymbolNames)
+                        {
+                            return false;
+                        }
+
+                        var excludedSymbols = operationBlockAnalysisContext.Options.GetExcludedSymbolNamesWithValueOption(SupportedDiagnostics[0], tree,
```

Comment:
We should instead be invoking the following extension method: https://github.com/dotnet/roslyn-analyzers/blob/641b5e8b6ddf8ccf095f5ac58a346ed1de546a18/src/Utilities/Compiler/Extensions/ISymbolExtensions.cs#L196-L204

Reply:
That's not the first time I am using the wrong method... I am wondering if we should hide `GetExcludedSymbolNamesWithValueOption` has a local function of the `IsConfiguredToSkipAnalysis` to avoid such issue. We could also create an analyzer but that seems overkill compared to hiding the function.

Reply:
I guess we need to create a ticket for the disposable rules: https://github.com/dotnet/roslyn-analyzers/blob/master/src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/DisposeAnalysis/DisposeAnalysis.DisposeDataFlowOperationVisitor.cs#L108 I have once again used the wrong helper.